internal/service/infra_provision/api_ui_join_token_resource_test.go

```diff
@@ -0,0 +1,251 @@
+package infra_provision_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/infra_provision"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+func TestAccUIJoinTokenResource_basic(t *testing.T) {
+	var resourceName = "bloxone_infra_join_token.test"
+	var v infra_provision.HostactivationJoinToken
+	name := acctest.RandomNameWithPrefix("jt")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccUIJoinTokenBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckUIJoinTokenExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					// Test Read Only fields
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "token_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "join_token"),
+					resource.TestCheckResourceAttr(resourceName, "status", "ACTIVE"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccUIJoinTokenResource_disappears(t *testing.T) {
+	resourceName := "bloxone_infra_join_token.test"
+	var v infra_provision.HostactivationJoinToken
+	name := acctest.RandomNameWithPrefix("jt")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckUIJoinTokenDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUIJoinTokenBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckUIJoinTokenExists(context.Background(), resourceName, &v),
+					testAccCheckUIJoinTokenDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccUIJoinTokenResource_Description(t *testing.T) {
+	var resourceName = "bloxone_infra_join_token.test_description"
+	name1 := acctest.RandomNameWithPrefix("jt")
+	name2 := acctest.RandomNameWithPrefix("jt")
```

Comment:
Both having the same value?

Reply:
No, both are random with the same prefix. Example `jt-sdfasd` and `jt-drewsv`